fe1-web/store/reducers/SocialReducer.ts

```diff
@@ -99,11 +99,31 @@ const socialSlice = createSlice({
         }
       },
     },
+
+    // Delete a chirp in the list of chirps
+    deleteChirp: {
+      prepare(laoId: Hash | string, chirp: ChirpState): any {
+        return { payload: { laoId: laoId.valueOf(), chirp: chirp } };
+      },
+      reducer(state, action: PayloadAction<{
+        laoId: string,
+        chirp: ChirpState,
+      }>) {
+        const { laoId, chirp } = action.payload;
+        const store = state.byLaoId[laoId];
+
+        if (store.byId[chirp.id]) {
+          if (!store.byId[chirp.id].isDeleted) {
+            store.byId[chirp.id] = chirp;
+          }
+        }
```

Comment:
```suggestion
        store.byId[chirp.id] = chirp;
```
wouldn't this be more robust / sufficient ?

Reply:
Also, thinking about it, should we actually consider setting the `isDeleted` and `text` fields here? 
It might make sense not to rely on the `chirp` parameter being configured correctly from the calling code.

Reply:
Alternatively, we could check in the ingestion that we "fail fast" on processing if we haven't received the chirp yet. 
In this case, your current approach makes a lot of sense, but then the ingestion should really fail so that the message remains marked as unprocessed.

Reply:
Final question, concerning the protocol, should a different sender (than the original chirp) be allowed to delete a chirp? And if not, shouldn't we check it here?

Reply:
> wouldn't this be more robust / sufficient ?
This enables marking a non-stored chirp as deleted to avoid reordering situation, then later if the reducer receives addChirp, since the chirp id already presents in byId, the store will not update its state. But it still wants to add it to the allIdsInOrder list and byUser map to display the deleted chirp. So, along with this, I will also need to move the insertion to allIdsInOrder and byUser outside the if(!store.byId[chirp.id]) in addChirp, or I still miss something?

Reply:
> Also, thinking about it, should we actually consider setting the `isDeleted` and `text` fields here? It might make sense not to rely on the `chirp` parameter being configured correctly from the calling code.

Can you elaborate on why we cannot trust it from the calling code (does it refer to deleteChirp(lao.id, chirp.toState()) in Socialhandler), please? I don't see what makes a difference in setting them in SocialHandler and in SocialReducer.

Reply:
> Final question, concerning the protocol, should a different sender (than the original chirp) be allowed to delete a chirp? And if not, shouldn't we check it here?

No, only the sender of the original chirp can delete a chirp. With the current UI, only the sender can send a delete request, but it's always safer to have a check on it. So if the chirp is present in `byId`, I compare the sender of the two. But how can I handle the case if the chirp is not yet stored in `byId`, in an out-of-order situation? And, if we cannot rely on the chirp parameter from the calling code as the same reason for i`sDeleted` and `text`, can I rely on the `sender`?